helpers/lib_tasks_docker_release.py

```diff
@@ -62,6 +62,278 @@ def _get_dev_version(version: str, container_dir_name: str) -> str:
+    """
+    hlitauti.run(ctx, cmd)
+
+
+def _list_image(ctx: Any, image_versioned: str) -> None:
+    """
+    List the image with the given version.
+
+    :param ctx: invoke context
+    :param image_versioned: image with the given version
+    """
+    cmd = f"docker image ls {image_versioned}"
+    hlitauti.run(ctx, cmd)
+
+
+def _run_tests(
+    ctx: Any,
+    stage: str,
+    version: str,
+    skip_tests: bool = False,
+    fast_tests: bool = False,
+    slow_tests: bool = False,
+    superslow_tests: bool = False,
+    qa_tests: bool = False,
+) -> None:
+    """
+    Run tests for a given stage and version.
+
+    :params ctx: invoke context
+    :params stage: image stage ("local", "dev", "prod")
+    :params version: version to test
+    :params skip_tests: skip all tests if True
+    :params fast_tests: run fast tests
+    :params slow_tests: run slow tests
+    :params superslow_tests: run superslow tests
+    :params qa_tests: run QA tests
+    """
+    if skip_tests:
+        _LOG.warning("Skipping all tests and releasing")
+        return
+    if fast_tests:
+        hlitapyt.run_fast_tests(ctx, stage=stage, version=version)
+    if slow_tests:
+        hlitapyt.run_slow_tests(ctx, stage=stage, version=version)
+    if superslow_tests:
+        hlitapyt.run_superslow_tests(ctx, stage=stage, version=version)
+    if qa_tests:
+        hlitapyt.run_qa_tests(ctx, stage=stage, version=version)
+
+
+def _push_image(
+    ctx: Any,
+    image_version: str,
+    stage: str,
+    container_dir_name: str = ".",
+    push_to_repo: bool = True,
+) -> None:
+    """
+    Push an image to the registry.
+
+    :param ctx: invoke context
+    :param image_version: image version
+    :param stage: image stage ("dev" or "prod")
+    :param container_dir_name: directory where Dockerfile is located
+    :param push_to_repo: push to registry if True
+    """
+    if push_to_repo:
+        docker_push_prod_image(
+            ctx, version=image_version, container_dir_name=container_dir_name
+        )
+    else:
+        _LOG.warning("Skipping pushing %s image to repo as requested", stage)
+
+
+def _docker_tag_and_push_multi_arch_image(
+    ctx: Any,
+    version: str,
+    base_image: str,
+    target_registry: str,
+    container_dir_name: str,
+    stage: str,
+    target_stage: str,
+) -> None:
+    """
+    Tag and push a multi-arch image to the target registry using `docker buildx
+    imagetools`.
+
+    :params ctx: invoke context
+    :params version: version to tag the image with
+    :params base_image: base name of the image
+    :params target_registry: target Docker registry to push to (e.g.,
+        "aws_ecr.ck" or "dockerhub.causify")
+    :params container_dir_name: directory where Dockerfile is located
+    :params stage: source stage of the image ("local" or "prod")
+    :params target_stage: target stage to push the image as ("dev" or
+        "prod")
+    """
+    hlitauti.report_task(container_dir_name=container_dir_name)
```

Comment:
Let's move it back to the caller since this is the helper function and not a task.